tools/install.sh

```diff
@@ -369,25 +376,44 @@ if [ -z "$KING_PHISHER_SKIP_SERVER" ]; then
 	if [ "$KING_PHISHER_USE_POSTGRESQL" == "yes" ]; then
 		if [ -f "/etc/init.d/postgresql" ]; then
 			/etc/init.d/postgresql start &> /dev/null
+			service postgresql status | grep online | grep -v grep &> /dev/null
+			if [ ! $? -eq 0 ]; then
+				postgresql-setup --initdb &> /dev/null
+				/etc/init.d/postgresql start &> /dev/null
+			fi
+		fi
+		if [ -f "/usr/lib/systemd/system/postgresql.service" ]; then
+			systemctl start postgresql &> /dev/null
+			systemctl is-active postgresql | grep active | grep -v grep &> /dev/null
+			if [ ! $? -eq 0 ]; then
+				postgresql-setup --initdb &> /dev/null
+				systemctl start postgresql &> /dev/null
+			fi
 		fi
 		echo "Configuring the PostgreSQL server"
 		PG_CONFIG_LOCATION=$(su postgres -c "psql -t -P format=unaligned -c 'show hba_file';")
 		echo "PostgreSQL configuration file found at $PG_CONFIG_LOCATION"
-		echo "# permit local connections from the king_phisher user for the king_phisher database" >> $PG_CONFIG_LOCATION
-		echo "host     king_phisher    king_phisher   127.0.0.1/32            md5" >> $PG_CONFIG_LOCATION
+		# put the king_phisher first in the line for localhost connects with md5
+		sed -i '/# IPv4 local connections:/a\# permit local connections from the king_phisher user for the king_phisher database' $PG_CONFIG_LOCATION
+		sed -i '/# permit local connections from the king_phisher user for the king_phisher database/a\host    king_phisher    king_phisher    127.0.0.1/32            md5' $PG_CONFIG_LOCATION
 		# generate a random 32 character long password for postgresql
 		PG_KP_PASSWORD=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' |  head -c 40)
 		su postgres -c "psql -c \"CREATE USER king_phisher WITH PASSWORD '$PG_KP_PASSWORD';\"" &> /dev/null
 		su postgres -c "psql -c \"CREATE DATABASE king_phisher OWNER king_phisher;\"" &> /dev/null
 		sed -i -re "s|database: sqlite://|#database: sqlite://|" ./server_config.yml
 		sed -i -re "s|#\\s?database: postgresql://.*$|database: postgresql://king_phisher:$PG_KP_PASSWORD@localhost/king_phisher|" ./server_config.yml
+		# restart postgresql to have hda_file updates take affect
+		if [ -f "/etc/init.d/postgresql" ]; then
+			/etc/init.d/postgresql restart &> /dev/null
```

Comment:
As per upstart it depands on the service itself of how it is utilized.
Currently Backbox is the only remaining distribution King Phisher supports that has not moved to systemd.

postgresql with backbox requires restart with `service postgresql <action>` or `/etc/init.d/postgresql <action>` See below for the out put.

```
root@backbox:~# restart postgresql
restart: Unknown job: postgresql
root@backbox:~# service postgresql
Usage: /etc/init.d/postgresql {start|stop|restart|reload|force-reload|status} [version ..]
root@backbox:~# service postgresql restart
 * Restarting PostgreSQL 9.3 database server                                                                                                                                                                                                                          [ OK ] 
root@backbox:~# /etc/init.d/postgresql restart
 * Restarting PostgreSQL 9.3 database server                                                                                                                                                                                                                          [ OK ] 
root@backbox:~# 
```